contracts/TokenOracle.sol

```diff
@@ -0,0 +1,31 @@
+pragma solidity ^0.4.24;
+
+import "@aragon/os/contracts/apps/AragonApp.sol";
+import "@aragon/os/contracts/acl/IACLOracle.sol";
+import "@aragon/os/contracts/lib/token/ERC20.sol";
+
+
+contract TokenOracle is AragonApp, IACLOracle {
+
+    bytes32 public constant CHANGE_TOKEN_ROLE = keccak256("CHANGE_DURATION_ROLE");
+
+    ERC20 public token;
+
+    function initialize(address _token) external onlyInit {
+        initialized();
+
+        token = ERC20(_token);
+    }
+
+    function changeToken(address _token) external auth(CHANGE_TOKEN_ROLE) {
+        token = ERC20(_token);
+    }
+
+    /**
+    * @notice ACLOracle
+    * @dev IACLOracle interface conformance
+    */
+    function canPerform(address _sender, address, bytes32, uint256[]) external view returns (bool) {
```

Comment:
Note that the `how` would be the how of the `authP` modifier of the role (e.g. `authP(TRANSFER_TOKENS, arr(receiver, amount))`). Per the example, what you'd be able to access in `how` here would be `[receiver, amount]`

Reply:
Ah yeah, it might be useful to add the account we wish to check the balance of as well. I think this means the function with the `authP` could then be called by a forwarder, provided that function verify the address that is passed to `authP` somehow. Eg the signing approach from the Redemptions app or a verified address array.

@lkngtn I'm pretty sure this merges the two approaches we discussed for the Dissent Oracle.